### PR TITLE
Add deployment timeline.

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

**Comment by danielballan** [comment]
This is derived from https://paper.dropbox.com/doc/Deployment-19-2-debrief-2019-04-30--ActVYsVgd8WoQofh3MSHiJxdAg-6JBT9wPTU0o1CnToznqN9

I have imposed some editing around the wording but have kept to the timeline agreed to by the group. I added one additional deadline, the first one where we agree on what we think we can get merged for a release. I think it's important to have that conversation in advance so we aren't arguing about it during the actual release process. We had discussed this in oral conversations about the deployment timeline even though it didn't make it into the minutes.

File: source/deployment/timeline.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* All packages planned for release have had release notes written and tags
+  pushed.
+* The project-wide documentation has been reviewed and any open issues closed
+  or intentionally deferred.
+
+T - 3 Weeks
+===========
+
+* All of the packages, including beamline-specific ones, have been built.
+* The open GitHub issues on lightsource-recipes have been addressed or
```
Comment (mrakitin) [inline source/deployment/timeline.rst]:
```suggestion
* The open GitHub issues and PRs on `lightsource2-recipes <https://github.com/NSLS-II/lightsource2-recipes>`_ have been addressed or
```

**Review by awalter-bnl — APPROVED** [review]
looks good to me

File: source/deployment/timeline.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+* All packages planned for release have had release notes written and tags
+  pushed.
+* The project-wide documentation has been reviewed and any open issues closed
+  or intentionally deferred.
+
+T - 3 Weeks
+===========
+
+* All of the packages, including beamline-specific ones, have been built.
```
Comment (mrakitin) [inline source/deployment/timeline.rst:31]:
```suggestion
* The deployment/release manager gathers requirements about beamline-specific packages.
* All of the packages, including beamline-specific ones, have been built.
```

File: source/deployment/timeline.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* The open GitHub issues on lightsource-recipes have been addressed or
+  intentionally deferred.
+
+T - 2 Weeks
+===========
+
+* Environments have been built and reproducible environments exported.
+* The playbooks repo has been updated to include new environments and archive
```
Comment (mrakitin) [inline source/deployment/timeline.rst]:
```suggestion
* The `playbooks <https://github.com/NSLS-II/playbooks>`_ repo has been updated to include new environments and archive
```

**Review by mrakitin — COMMENTED** [review]
Added a couple of suggestions and edits.

**Review by mrakitin — APPROVED** [review]
Looks good to me now. Thanks.

File: source/deployment/timeline.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* Review open PRs and issues on all projects. Agree on what will and will not
+  make it into the release, with 1.5 weeks to push anything over the finishline
+  that just needs final touches.
+
+T - 4.5 Weeks
+=============
+
+* Defer any unfinished PRs and begin release process.
+
+T - 4 Weeks
```
Comment (ke-zhang-rd) [inline source/deployment/timeline.rst:19]:
```suggestion
T - 4 Weeks, Jun 3rd
```

**Comment by danielballan** [comment]
I see 14fdbc0 added a date next to "T - 6 Weeks". I had in mind that this would be a generic reference for all future cycles, not this upcoming one specifically, which is why I did not include specific dates. Should we add dates and update them for each cycle? If so, we should add them to _all_ the headings, not just T - 6.

**Comment by danielballan** [comment]
I see a pushed commit adding a date to "T - 6" and a suggestion adding a date to "T - 4". Would you add suggestions for the other "T - ..." headings as well?

File: source/deployment/timeline.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+`NSLS-II Operating Schedule <https://www.bnl.gov/ps/nsls2/opschedule.php>`_.
+
+T - 6 Weeks, May 20th
+=====================
+
+* Review open PRs and issues on all projects. Agree on what will and will not
+  make it into the release, with 1.5 weeks to push anything over the finishline
+  that just needs final touches.
+
+T - 4.5 Weeks
```
Comment (ke-zhang-rd) [inline source/deployment/timeline.rst:14]:
```suggestion
T - 4.5 Weeks, May 29th
```

File: source/deployment/timeline.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+T - 4 Weeks
+===========
+
+* All packages planned for release have had release notes written and tags
+  pushed.
+* The project-wide documentation has been reviewed and any open issues closed
+  or intentionally deferred.
+
+T - 3 Weeks
```
Comment (ke-zhang-rd) [inline source/deployment/timeline.rst:27]:
```suggestion
T - 3 Weeks, Jun 10th
```

File: source/deployment/timeline.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+===========
+
+* The deployment/release manager gathers requirements about beamline-specific packages.
+* All of the packages, including beamline-specific ones, have been built.
+* The open GitHub issues and PRs on `lightsource2-recipes <https://github.com/NSLS-II/lightsource2-recipes>`_ have been addressed or
+  intentionally deferred.
+
+T - 2 Weeks
```
Comment (ke-zhang-rd) [inline source/deployment/timeline.rst:35]:
```suggestion
T - 2 Weeks, Jun 17th
```

File: source/deployment/timeline.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+===========
+
+* Environments have been built and reproducible environments exported.
+* The `playbooks <https://github.com/NSLS-II/playbooks>`_ repo has been updated to include new environments and archive
+  old ones.
+
+T - 1 Week
```
Comment (ke-zhang-rd) [inline source/deployment/timeline.rst:42]:
```suggestion
T - 1 Week, Jun 24th
```

**Comment by awalter-bnl** [comment]
I think I agree with @danielballan that this should not include any dates so that it can easily be reviewed. I think that those dates should be placed into the DAMA google calender, as they have been.

**Comment by mrakitin** [comment]
Can we merge this PR or do we think we need more work on it?

**Comment by awalter-bnl** [comment]
@mrakitin , I think @danielballan requested that the dates be removed first

**Comment by danielballan** [comment]
Yes. If we agree with @awalter-bnl above (I do) then 14fdbc0 needs to be force-pushed off.

**Comment by mrakitin** [comment]
I agree. We are keeping track of dater in DAMA calendar. That way we don't have to update dates each cycle in the docs.

**Comment by danielballan** [comment]
OK, I force-pushed off the commit that had added some dates.

**Comment by mrakitin** [comment]
Thank you!

**Pull request merged**